api/hypershift/v1beta1/hostedcluster_types.go

```diff
@@ -1260,9 +1260,101 @@ const (
+	// +optional
+	DelayAfterDeleteSeconds *int32 `json:"delayAfterDeleteSeconds,omitempty"`
+
+	// delayAfterFailureSeconds sets how long after a scale down failure, scale down evaluation resumes.
+	// It must be between 0 and 86400 (24 hours).
+	// When set to 0, this means scale down evaluation will resume immediately after a scale down failure, without any delay.
+	// When omitted, the autoscaler defaults to 180s (3 minutes).
+	//
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:validation:Maximum=86400
+	// +optional
+	DelayAfterFailureSeconds *int32 `json:"delayAfterFailureSeconds,omitempty"`
+
+	// unneededDurationSeconds establishes how long a node should be unneeded before it is eligible for scale down in seconds.
+	// It must be between 0 and 86400 (24 hours).
+	// When omitted, the autoscaler defaults to 600s (10 minutes).
+	//
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:validation:Maximum=86400
+	// +optional
+	UnneededDurationSeconds *int32 `json:"unneededDurationSeconds,omitempty"`
+
+	// utilizationThresholdPercent determines the node utilization level, defined as sum of requested resources divided by capacity, below which a node can be considered for scale down.
+	// The value represents a percentage from 0 to 100.
+	// When set to 0, this means nodes will only be considered for scale down if they are completely idle (0% utilization).
+	// When set to 100, this means nodes will be considered for scale down regardless of their utilization level.
+	// A value between 0 and 100 represents the utilization threshold below which a node can be considered for scale down.
+	// When omitted, the autoscaler defaults to 50%.
+	//
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:validation:Maximum=100
+	// +optional
+	UtilizationThresholdPercent *int32 `json:"utilizationThresholdPercent,omitempty"`
+}
+
+// ExpanderString contains the name of an expander to be used by the cluster autoscaler.
+// +kubebuilder:validation:Enum=LeastWaste;Priority;Random
+type ExpanderString string
+
+// These constants define the valid values for an ExpanderString
+const (
+	LeastWasteExpander ExpanderString = "LeastWaste" // Selects the node group with the least idle resources.
+	PriorityExpander   ExpanderString = "Priority"   // Selects the node group with the highest priority.
+	RandomExpander     ExpanderString = "Random"     // Selects a node group randomly.
+)
+
 // ClusterAutoscaling specifies auto-scaling behavior that applies to all
 // NodePools associated with a control plane.
 type ClusterAutoscaling struct {
+	// scaling defines the scaling behavior for the cluster autoscaler.
+	// ScaleUpOnly means the autoscaler will only scale up nodes, never scale down.
+	// ScaleUpAndScaleDown means the autoscaler will both scale up and scale down nodes.
+	// When set to ScaleUpAndScaleDown, the scaleDown field can be used to configure scale down behavior.
+	//
+	// +kubebuilder:default=ScaleUpAndScaleDown
```

Comment:
Is this the correct default?

Reply:
the default behaviour for this api needs to be the one for already running clusters, which is scale up and down. I think the question is whether we want to convey that by defaulting here or by letting empty be it

Reply:
Indeed, you could have this omitted completely (which means no change from a readers perspective after this change).

This would in theory allow you to change the default under the hood in the future if our opinions of best practice changed.

Those that had set an explicit value would of course not be affected.

I could however see an argument that this _should_ be defaulted in this way, since we don't want to change the scaling behaviour of an existing cluster maybe?

Reply:
I think we should keep the explicit `+kubebuilder:default=ScaleUpAndScaleDown` for clarity and consistency. 

The current default behavior for existing clusters is scale up and down, and making this explicit in the API (rather than hiding it in implementation) provides better user experience and documentation. It also ensures forward compatibility for existing clusters.

Reply:
> the default behaviour for this api needs to be the one for already running clusters, which is scale up and down

@enxebre I am little confused by this comment. Is the scaling down behavior determined by the arg: `--scale-down-enabled`  in the `cluster-autoscaler` pod ?  or hypershift has it's own logic besides of this argument ? 

Now before this feature gets implemented, as long as the  `autoScaling`  in the nodepool gets enabled  without setting `autoscaling` in the hosted cluster, the `cluster-autoscaler` pod will be created with the default arguments, where I don't see the scaling down argument.